internal/provider/resource_gitlab_runner.go

```diff
@@ -0,0 +1,257 @@
+		// Since this resource updates an in-place resource, the update method is the same as the create method
+		CreateContext: resourceGitLabRunnerCreate,
+		UpdateContext: resourceGitLabRunnerUpdate,
+		ReadContext:   resourceGitLabRunnerRead,
+		DeleteContext: resourceGitLabRunnerDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"token": {
+				Description: `The registration token used to register the runner`,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Sensitive:   true,
+			},
+			// Even though the output variable is just called 'token', we need to differentiate between the regitration
+			// and authentication token in terraform.
+			"authentication_token": {
+				Description: `The authentication token used for building a config.toml file. This value is not present when imported.`,
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"status": {
+				Description: `The status of runners to show, one of: online and offline. active and paused are also possible values
+				 which were deprecated in GitLab 14.8 and will be removed in GitLab 16.0`,
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Description: `The runner's description'`,
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"paused": {
+				Description: `Whether the runner should ignore new jobs`,
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     gitlab.Bool(false),
+			},
+			"locked": {
+				Description: `Whether the runner should be locked for current project`,
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     gitlab.Bool(false),
+			},
+			"run_untagged": {
+				Description: `Whether the runner should handle untagged jobs`,
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     gitlab.Bool(true),
+			},
+			"tag_list": {
+				Description: `List of runner’s tags`,
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"access_level": {
+				Description: fmt.Sprintf(`The access_level of the runner. Valid values are: %s`,
+					renderValueListForDocs(runnerAccessLevelAllowedValues)),
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(runnerAccessLevelAllowedValues, true),
+				Default:      "not_protected",
+			},
+			"maximum_timeout": {
+				Description: `Maximum timeout set when this runner handles the job`,
+				Type:        schema.TypeInt,
+				Optional:    true,
+			},
+			// While Maintenance Note is available during "create", it is not available during "update", so
+			// excluding it here for now.
+		},
+	}
+})
+
+func resourceGitLabRunnerDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Any reason for the delete first?

NIT: We usually had it in `CRUD` order :)

Reply:
No reason, just the order I created them in 😄 . Updated!